paddle/majel/allocation.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+#include <memory>
+
+#include <majel/place.h>
+#include "hl_gpu.h"
```

Comment:
`hl_gpu.h` is a "hub" file, which includes all other header files. Such kind of hub header files should be discouraged, as it prevents us from making module dependencies clear.